AutomationHelpers/src/UserCodeCollections/FileLibrary.cs

```diff
@@ -152,6 +153,148 @@ public static void WaitForFile(string path, string pattern, int duration, int in
+                            Report.Error("Files '" + filePath1 + "' and '" + filePath2 + "' are not equal.");
+                            return;
+                        }
+                    }
+                }
+
+                Report.Success("Files '" + filePath1 + "' and '" + filePath2 + "' are equal.");
+            }
+            catch (Exception ex)
+            {
+                Report.Error(ex.Message);
+            }
+        }
+
+        /// <summary>
+        /// Compares content of two text files.
+        /// </summary>
+        /// <param name="filePath1">The relative or absolute path of the first file</param>
+        /// <param name="filePath2">The relative or absolute path of the second file</param>
+        [UserCodeMethod]
+        public static void CompareTextFiles(string filePath1, string filePath2)
+        {
+            try
+            {
+                filePath1 = GetPathForFile(filePath1);
+                filePath2 = GetPathForFile(filePath2);
+
+                if (!File.Exists(filePath1))
+                {
+                    Report.Failure("File '" + filePath1 + "' does not exist.");
+                    return;
+                }
+
+                if (!File.Exists(filePath2))
+                {
+                    Report.Failure("File '" + filePath2 + "' does not exist.");
+                    return;
+                }
+
+                var fileContent1 = File.ReadAllText(filePath1);
+                fileContent1 = Regex.Replace(fileContent1, "(\r\n)|(\n\r)|(\n)|(\r)", "\r\n");
+                var fileContent2 = File.ReadAllText(filePath2);
+                fileContent2 = Regex.Replace(fileContent2, "(\r\n)|(\n\r)|(\n)|(\r)", "\r\n");
+
+                if (fileContent1 != fileContent2)
+                {
+                    Report.Error("Files '" + filePath1 + "' and '" + filePath2 + "' are not equal.");
+                    return;
+                }
+
+                Report.Success("Files '" + filePath1 + "' and '" + filePath2 + "' are equal.");
+            }
+            catch (Exception ex)
+            {
+                Report.Error(ex.Message);
+            }
+        }
+
+        /// <summary>
+        /// Checks if file contains text specified.
+        /// </summary>
+        /// <param name="filePath">The relative or absolute path to the file</param>
+        /// <param name="text">The text to search for</param>
+        [UserCodeMethod]
+        public static void SearchText(string filePath, string text)
+        {
+            try
+            {
+                filePath = GetPathForFile(filePath);
+
+                if (!File.Exists(filePath))
+                {
+                    Report.Error("File '" + filePath + "' does not exists.");
+                    return;
+                }
+
+                if (!File.ReadAllText(filePath).Contains(text))
+                {
+                    Report.Error("File '" + filePath + "' contains text '" + text + "'.");

```

Comment:
If check passes when file doesn't contain text, and then adding error to report saying that it does contain text?